src/persistence/paths.cpp

```diff
@@ -0,0 +1,243 @@
+#include "paths.h"
+
+#include <QApplication>
+#include <QDebug>
+#include <QDir>
+#include <QFile>
+#include <QStandardPaths>
+#include <QString>
+#include <QStringBuilder>
+#include <QStringList>
+
+namespace {
+const QLatin1Literal globalSettingsFile{"qtox.ini"};
+const QLatin1Literal profileFolder{"profiles"};
+const QLatin1Literal themeFolder{"themes"};
+const QLatin1Literal avatarsFolder{"avatars"};
+const QLatin1Literal transfersFolder{"transfers"};
+const QLatin1Literal screenshotsFolder{"screenshots"};
+
+// NOTE(sudden6): currently unused, but reflects the TCS at 2018-11
+#ifdef Q_OS_WIN
+const QLatin1Literal TCSToxFileFolder{"%APPDATA%/tox/"};
+#elif defined(Q_OS_OSX)
+const QLatin1Literal TCSToxFileFolder{"~/Library/Application Support/Tox"};
+#else
+const QLatin1Literal TCSToxFileFolder{"~/.config/tox/"};
+#endif
+} // namespace
+
+/**
+ * @class Profile
+ * @brief Handles all qTox internal paths
+ *
+ * The qTox internal file layout starts at `<BASE_PATH>`. This directory is platform
+ * specific and depends on if qTox runs in portable mode.
+ *
+ * Example file layout for non-portable mode:
+ * @code
+ *  <BASE_PATH>/themes/
+ *             /profiles/
+ *             /...
+ * @endcode
+ *
+ * Example file layout for portable mode:
+ * @code
+ *  <BASE_PATH>/themes/
+ *             /profiles/
+ *             /qtox.ini
+ * @endcode
+ *
+ * All qTox or Tox specific directories should be looked up through this module.
+ */
+
+/**
+ * @brief Paths::makePaths Factory method for the Paths object
+ * @param forcePortable if true, force portable mode
+ * @param forceNonPortable if true, force non-portable mode
+ * @return Pointer to Paths object on success, nullptr else
+ * @note forcePortable and forceNonPortable can't be true at the same time
+ *       if both of them are false, we autodetect if portable mode should be used
+ */
+Paths* Paths::makePaths(bool forcePortable, bool forceNonPortable)
```

Comment:
Two conflicting bools in parameters looks strange. What about enum?
`enum Path::Portable { Auto, Portable, NonPortable };`

Reply:
I thought about it, but decided for standard types, will change.